src/test/storage/iterables_test.cpp

```diff
@@ -103,139 +108,197 @@ class IterablesTest : public BaseTest {
+INSTANTIATE_TEST_SUITE_P(SegmentEncoding, EncodedSegmentIterablesTest,
+                         ::testing::Combine(::testing::ValuesIn(all_segment_encoding_specs),
+                                            ::testing::Bool(),   // nullable
+                                            ::testing::Bool()),  // position filter
+                         formatter);
+
+TEST_P(EncodedSegmentIterablesTest, IteratorWithIterators) {
+  auto encoding_spec = std::get<0>(GetParam());
+  auto nullable = std::get<1>(GetParam());
+  auto with_position_filter = std::get<2>(GetParam());
+  std::shared_ptr<Table> test_table = (nullable ? table : table_with_null);
+
+  auto chunk_encoding_spec = ChunkEncodingSpec{test_table->column_count(), EncodingType::Unencoded};
+  for (auto column_id = ColumnID{0}; column_id < test_table->column_count(); ++column_id) {
+    if (encoding_supports_data_type(encoding_spec.encoding_type, test_table->column_data_type(column_id))) {
+      chunk_encoding_spec[column_id] = encoding_spec;
+    }
+  }
+  ChunkEncoder::encode_all_chunks(test_table, chunk_encoding_spec);
+
+  auto chunk = test_table->get_chunk(ChunkID{0u});
+  auto base_segment = chunk->get_segment(ColumnID{0u});
+
+  resolve_data_and_segment_type(*base_segment, [&](const auto data_type_t, const auto& segment) {
+    using ColumnDataType = typename decltype(data_type_t)::type;
+    using SegmentType = std::decay_t<decltype(segment)>;
+
+    if constexpr (!std::is_same_v<pmr_string, ColumnDataType>) {
+      auto sum = ColumnDataType{0};
+      auto accessed_offsets = std::vector<ChunkOffset>{};
+      auto functor = SumUpWithIterator<ColumnDataType>{sum, accessed_offsets};
+
+      const auto iterable = create_iterable_from_segment<ColumnDataType, false /* no type erasure */>(segment);
+      if (with_position_filter) {
+        if constexpr (!std::is_same_v<SegmentType, ReferenceSegment>) {
+          iterable.with_iterators(position_filter, functor);
+        }
+      } else {
+        iterable.with_iterators(functor);
+      }
+
+      EXPECT_EQ(sum, expected_sum(nullable, with_position_filter));
+      EXPECT_EQ(accessed_offsets, expected_offsets(with_position_filter));
+    }
+  });
 }
 
-TEST_F(IterablesTest, ValueSegmentNullableReferencedIteratorWithIterators) {
-  const auto chunk = table_with_null->get_chunk(ChunkID{0u});
-
-  auto segment = chunk->get_segment(ColumnID{0u});
-  auto int_segment = std::dynamic_pointer_cast<const ValueSegment<int>>(segment);
-
-  auto iterable = ValueSegmentIterable<int>{*int_segment};
-
-  auto sum = uint32_t{0};
-  auto accessed_offsets = std::vector<ChunkOffset>{};
-  iterable.with_iterators(position_filter, SumUpWithIterator{sum, accessed_offsets});
+class EncodedStringSegmentIterablesTest : public IterablesTest,
+                                          public ::testing::WithParamInterface<std::tuple<SegmentEncodingSpec, bool>> {
```

Comment:
Better use `BaseTestWithParams`.

Reply:
how can I do so? I think, we need the interface here, since IterablesTest is also derived from BaseTest and using BaseTestWithParams as second base class causes ambiguities.

Reply:
If it's ot straightforward, just ignore my comment. ;)